src/component/contents/exploration/DraftEditorBlockNode.react.js

```diff
@@ -0,0 +1,370 @@
+const DraftOffsetKey = require('DraftOffsetKey');
+const EditorState = require('EditorState');
+const Immutable = require('immutable');
+const React = require('React');
+const ReactDOM = require('ReactDOM');
+const Scroll = require('Scroll');
+const Style = require('Style');
+const getElementPosition = require('getElementPosition');
+const getScrollPosition = require('getScrollPosition');
+const getViewportDimensions = require('getViewportDimensions');
+const invariant = require('invariant');
+
+const SCROLL_BUFFER = 10;
+
+const {List} = Immutable;
+
+type BlockRenderFn = (block: BlockNodeRecord) => ?Object;
+type BlockStyleFn = (block: BlockNodeRecord) => string;
+
+type Props = {
+  block: BlockNodeRecord,
+  blockProps?: Object,
+  blockRenderMap: DraftBlockRenderMap,
+  blockRendererFn: BlockRenderFn,
+  blockStyleFn: BlockStyleFn,
+  contentState: ContentState,
+  customStyleFn: (style: DraftInlineStyle, block: BlockNodeRecord) => ?Object,
+  customStyleMap: Object,
+  decorator: ?DraftDecoratorType,
+  direction: BidiDirection,
+  editorKey: string,
+  editorState: EditorState,
+  forceSelection: boolean,
+  selection: SelectionState,
+  startIndent?: boolean,
+  tree: List<any>,
+};
+
+/**
+ * Return whether a block overlaps with either edge of the `SelectionState`.
+ */
+const isBlockOnSelectionEdge = (
+  selection: SelectionState,
+  key: string,
+): boolean => {
+  return selection.getAnchorKey() === key || selection.getFocusKey() === key;
+};
+
+/**
+ * We will use this helper to identify blocks that need to be wrapped but have siblings that
+ * also share the same wrapper element, this way we can do the wrapping once the last sibling
+ * is added.
+ */
+const shouldNotAddWrapperElement = (
+  block: BlockNodeRecord,
+  contentState: ContentState,
+): boolean => {
+  const nextSiblingKey = block.getNextSiblingKey();
+
+  return nextSiblingKey
+    ? contentState.getBlockForKey(nextSiblingKey).getType() === block.getType()
+    : false;
+};
+
+const applyWrapperElementToSiblings = (
+  wrapperTemplate: *,
+  Element: string,
+  nodes: Array<React.Node>,
+): Array<React.Node> => {
+  const wrappedSiblings = [];
+
+  // we check back until we find a sibbling that does not have same wrapper
+  for (const sibling: any of nodes.reverse()) {
+    if (sibling.type !== Element) {
+      break;
+    }
+    wrappedSiblings.push(sibling);
+  }
+
+  // we now should remove from acc the wrappedSiblings and add them back under same wrap
+  nodes.splice(nodes.indexOf(wrappedSiblings[0]), wrappedSiblings.length + 1);
+
+  const childrenIs = wrappedSiblings.reverse();
+
+  const key = childrenIs[0].key;
+
+  nodes.push(
+    React.cloneElement(
+      wrapperTemplate,
+      {
+        key: `${key}-wrap`,
+        'data-offset-key': DraftOffsetKey.encode(key, 0, 0),
+      },
+      childrenIs,
+    ),
+  );
+
+  return nodes;
+};
+
+const getDraftRenderConfig = (
+  block: BlockNodeRecord,
+  blockRenderMap: DraftBlockRenderMap,
+): * => {
```

Comment:
We could at least say it returns type `Object` and leave a TODO to add a specific type. Same below with `getCustomRenderConfig`.

Reply:
Sounds good to me, I was leaning towards adding typing now but the PR was already too big, happy to do this on follow up PR's